seatunnel-connectors-v2/connector-jdbc/src/main/java/org/apache/seatunnel/connectors/seatunnel/jdbc/utils/JdbcCatalogUtils.java

```diff
@@ -409,4 +423,101 @@ private static ReadonlyConfig extractCatalogConfig(JdbcConnectionConfig config)
         catalogConfig.put(JdbcOptions.HANDLE_BLOB_AS_STRING.key(), config.isHandleBlobAsString());
         return ReadonlyConfig.fromMap(catalogConfig);
     }
+
+    /** Process table path with regex pattern and add matched tables to the result. */
+    private static void processRegexTablePath(
+            AbstractJdbcCatalog jdbcCatalog,
+            JdbcDialect jdbcDialect,
+            JdbcSourceTableConfig tableConfig,
+            Map<TablePath, JdbcSourceTable> result)
+            throws SQLException {
+
+        String tablePath = tableConfig.getTablePath();
+        log.info("Processing table path with regex: {}", tablePath);
+
+        String databasePattern = ".*";
+        String tableNamePattern = ".*";
+
+        String processedTablePath = tablePath.replace("\\.", DOT_PLACEHOLDER);
+        log.debug("After replacing escaped dots with placeholder: {}", processedTablePath);
+
+        String[] parts = processedTablePath.split("\\.");
+
+        String fullTablePattern;
+
+        boolean isOracleDialect =
+                jdbcDialect.dialectName().equalsIgnoreCase("oracle")
+                        || jdbcDialect.dialectName().equalsIgnoreCase("oceanbase-oracle");
```

Comment:
We cannot add this special processing logic. There are more than two Oracle-like databases.